pkg/traversal/worker.go

```diff
@@ -159,42 +145,77 @@ func (w *worker) shouldOmitK8sResource(resource *kube.ResourceList) (bool, error
 	return false, nil
 }
 
-func (w *worker) obfuscateFile(wf workerFile, relativePathToFile string) error {
-	closeWriter, writer, err := w.writer.Writer(relativePathToFile, wf.f.Permissions())
+func (w *Worker) obfuscateFileContent(inputFile string, outputFile string) error {
+	readPath := filepath.Join(w.inputFolder, inputFile)
+	writePath := filepath.Join(w.outputFolder, outputFile)
+	writePathParentDir := filepath.Dir(writePath)
+
+	inputOsFile, err := os.Open(readPath)
 	if err != nil {
 		return err
 	}
-	// close the output file when done
-	defer func() {
-		if err := closeWriter(); err != nil {
-			w.errorCh <- &fileProcessingError{
-				path:  wf.f.Path(),
-				cause: err,
-			}
-		}
-	}()
+	scanner := bufio.NewScanner(inputOsFile)
+
+	err = os.MkdirAll(writePathParentDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create directory %s: %w", writePathParentDir, err)
+	}
 
-	scanner, closeReader, err := wf.f.Scanner()
+	outputOsFile, err := os.OpenFile(writePath, os.O_CREATE|os.O_WRONLY, 0666)
 	if err != nil {
 		return err
 	}
-	defer func() {
-		if err := closeReader(); err != nil {
-			w.errorCh <- &fileProcessingError{
-				path:  wf.f.Path(),
-				cause: err,
-			}
-		}
-	}()
+
+	writer := bufio.NewWriter(outputOsFile)
 	for scanner.Scan() {
 		contents := scanner.Text()
 		for _, o := range w.obfuscators {
 			contents = o.Contents(contents)
 		}
-		_, err = writer.WriteString(fmt.Sprintf("%s\n", contents))
+
+		_, err = fmt.Fprintln(writer, contents)
 		if err != nil {
 			return err
 		}
 	}
+
+	// we deliberately do not defer the closes, as this will be processed asynchronous and
```

Comment:
This comment does not make clear to anyone reading the code. Because returning an error makes the error reader worker call `klog.Exitf()` and hence all file handles will immediately be cleaned up.

Reply:
that's the whole point and hence the explanation of why there is no usage of defer for closing the handles. How would you write that in a comment?

Reply:
(removed the comment now)